pkg/apis/autopilot.k0sproject.io/v1beta2/updateconfig.go

```diff
@@ -40,10 +40,43 @@ type UpdateConfig struct {
 }
 
 type UpdateSpec struct {
-	Channel         string          `json:"channel,omitempty"`
-	UpdateServer    string          `json:"updateServer,omitempty"`
-	UpgradeStrategy UpgradeStrategy `json:"upgradeStrategy,omitempty"`
-	PlanSpec        PlanSpec        `json:"planSpec,omitempty"`
+	Channel         string            `json:"channel,omitempty"`
+	UpdateServer    string            `json:"updateServer,omitempty"`
+	UpgradeStrategy UpgradeStrategy   `json:"upgradeStrategy,omitempty"`
+	PlanSpec        AutopilotPlanSpec `json:"planSpec,omitempty"`
```

Comment:
Was PlanSpec renamed to AutopilotPlanSpec? Do we still use the old definition from file types.go (https://github.com/makhov/k0s/blob/update-server-plan-spec-improvements/pkg/apis/autopilot.k0sproject.io/v1beta2/types.go)?

Reply:
Yes, we do. Before the change, the `PlanSpec` was used both for the `Plan` and `UpdateConfig` objects. I've created a separate type for the `UpdateConfig` because `Plan` requires some additional fields that we generate automatically